### DISCUSSION TITLE
flux1-dev-bnb-nf4-v2

### DISCUSSION THREAD

**Discussion by ETCore7** [Acly/krita-ai-diffusion#1221] [Ideas]
Forge uses flux1-dev-bnb-nf4-v2, which is (much) better than flux1-dev-bnb-nf4 that comes with krita-ai-diffusion. However, when using flux1-dev-bnb-nf4-v2 in Krita there is a long list of "error(s) in loading state_dict" and "size mismatch" and such. Earlier I saw the same with another model, but that got solved after running Forge once, the model got patched or something. Forge does not do this with flux1-dev-bnb-nf4-v2, so krita-ai-diffusion can't use it.

Also, after deleting flux1-dev-bnb-nf4-v2 (which I had renamed to flux1-dev-bnb-nf4) I expected to see the model to be missing in the server configuration, but it wasn't.

**Comment by Acly** [comment]
Just use the GGUF quants.

**Reply by SyamsQ** [reply]
please tutorial for use the GGUF quants at Krita

**Reply by ETCore7** [reply]
FYI, in Invoke the model (and others) can be downloaded from within the app. 

![image](https://github.com/user-attachments/assets/cd3d8ba2-c390-4be8-be8c-234de92d9900)

In Krita you may need to configure something like this:

![image](https://github.com/user-attachments/assets/12d80f08-3fc4-49f1-a7b2-f3b588013522)

... which are available in Invoke by default. I haven't tried it in Krita, Flux in Krita is a bit too slow on my PC, but you may need to make a custom config file.

![image](https://github.com/user-attachments/assets/55b57950-e369-462f-9d94-340ef6729877)

By the way, Juggernaut XL v9 provides good quality too, and you can use the model in Krita without extra hassle.

You can easily get Invoke and many other tools in [Pinokio](https://pinokio.computer/), installations are automated.

![image](https://github.com/user-attachments/assets/6f8ea5b2-da83-447a-9516-b477feacda1d)

Invoke and Forge are image generators, CogStudio produces animations (txt2vid, img2vid, vid2vid, extend), FluxGym is a lora maker for Flux, and there are audio/music generators like Meta, AudioPlus, RC Stable Audio Tools, and more.

In case somebody finds this post by accident and wants more, GPT4All is an app by NomicAI to run LLM's locally. You can find and install many models from within the app and edit system prompts.

![image](https://github.com/user-attachments/assets/e5d482f0-0f8f-42b8-be97-eef41c0d3669)

**Reply by SyamsQ** [reply]
I don't understand yet. What does invoke have to do with the GGUF Krita?

**Comment by ETCore7** [comment]
In Invoke you can see what settings and things are needed to get the model working in Krita. If that was not the reason for your question please ignore.

**Reply by SyamsQ** [reply]
can invoke work offline and do inpainting?
![image](https://github.com/user-attachments/assets/585674fd-f270-462f-a2da-454ffae68a4b)

**Comment by ETCore7** [comment]
Yes, all apps in Pinokio are for offline use, and Invoke has tools like inpaint. There is a detailed manual on how to do things. [Inpainting, Outpainting](https://support.invoke.ai/support/solutions/articles/151000096702-inpainting-outpainting-and-bounding-box)

**Comment by ETCore7** [comment]
PS, the interface can be confusing when you're new to Invoke, here are two short Youtube tutorials:
[Part 1 - Navigation & Settings](https://www.youtube.com/watch?v=W4nRM1iA9Rw)
[Part 2 - Image Generation, img2img, In/Outpainting](https://www.youtube.com/watch?v=cx7L-evqLPo)

**Reply by SyamsQ** [reply]
is this invoke free like Krita +Comfyui?
![image](https://github.com/user-attachments/assets/24c34e24-769e-4711-8802-d4285d93cbde)

**Comment by ETCore7** [comment]
Invoke is free for non-commercial use. No need to sign up, just install the app in [Pinokio](https://pinokio.computer/).

**Reply by SyamsQ** [reply]
Is using invoke better than Krita + Comfyui for inpainting?

**Comment by ETCore7** [comment]
When you are using the same model there is no difference. What interface is more convenient is a personal preference.

**Reply by SyamsQ** [reply]
I found many errors in Invoke. From not being able to insert some models and Lora, to not being able to generate images even with the inserted models. I was confused. On Discord, no one has answered since yesterday.
![Screenshot 2024-10-14 142337](https://github.com/user-attachments/assets/ddb2dfda-a867-4290-af42-d64b5041cb0a)
![Screenshot 2024-10-14 141654](https://github.com/user-attachments/assets/e5298e64-0d06-4921-a01e-a4cb0c9cd79e)
![Screenshot 2024-10-14 032249](https://github.com/user-attachments/assets/69c39513-fa43-48b6-b95f-bdef6898cad4)
![Screenshot 2024-10-14 031914](https://github.com/user-attachments/assets/6772fb79-7afa-49e1-898a-2c3c08d2d08f)
![Screenshot 2024-10-14 031825](https://github.com/user-attachments/assets/a8165a96-9e60-420e-8866-9cb820ca7b35)
![Screenshot 2024-10-14 031739](https://github.com/user-attachments/assets/22071b43-1a88-43b1-a2e1-2ce5ee11bf08)

**Reply by ETCore7** [reply]
Sorry, I can't help you with that. The starter models work for me, maybe custom models and loras need additional settings/configurations, but that's way beyond my understanding.
